### PR TITLE
Service of application event and notifications (both post and email notifications for app / respondents)

### PR DESCRIPTION
**Pull request by pete-lowe**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-1440 
https://tools.hmcts.net/jira/browse/PRL-1441
https://tools.hmcts.net/jira/browse/PRL-1214 
https://tools.hmcts.net/jira/browse/PRL-1439 
https://tools.hmcts.net/jira/browse/PRL-1443 
https://tools.hmcts.net/jira/browse/PRL-1362
https://tools.hmcts.net/jira/browse/PRL-1442

### Change description ###

"Service of application event and notifications (post and email)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java [hunk 1]
```diff
@@ -84,7 +85,7 @@ public class CallbackController {
     private final AllTabServiceImpl allTabsService;
     private final UserService userService;
     private final DocumentGenService documentGenService;
-
+    private final ServiceOfApplicationService serviceOfApplicationService;
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:88]: Do we need this ServiceOfApplicationService here?

File: src/main/java/uk/gov/hmcts/reform/prl/enums/serviceofapplication/TransferOfCaseToAnotherCourtEnum.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    @JsonValue
+    public String getDisplayedValue() {
+        return displayedValue;
+    }
+
+    @JsonCreator
+    public static TransferOfCaseToAnotherCourtEnum getValue(String key) {
+        return TransferOfCaseToAnotherCourtEnum.valueOf(key);
+    }
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/enums/serviceofapplication/TransferOfCaseToAnotherCourtEnum.java:25]: We dont need this as C49 requirement has been withdrawn

File: src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/serviceofapplication/OrdersToServeSA.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private final List<SpecialGuardianShipEnum>    specialGuardianShipOption;
+
+    private final List<NoticeOfProceedingsPartiesEnum> noticeOfProceedingsPartiesOption;
+    private final List<NoticeOfProceedingsNonPartiesEnum>    noticeOfProceedingsNonPartiesOption;
+
+    private final List<TransferOfCaseToAnotherCourtEnum> transferOfCaseToAnotherCourtOption;
+    private final List<AppointmentOfGuardianEnum>    appointmentOfGuardianOption;
+
+    private final List<NonMolestationEnum> nonMolestationOption;
+    private final List<OccupationEnum>    occupationOption;
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/serviceofapplication/OrdersToServeSA.java:51]: formatting issue through out this file. Space issue mainly

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/notify/CaseInviteEmail.java [hunk 4]
```diff
@@ -4,17 +4,44 @@
...
         this.respondentFullName = String.format("%s %s", party.getFirstName(), party.getLastName());
+        this.caseLink = caseLink;
+        this.citizenSignUpLink = citizenSignUpLink;
+        this.caseName = caseData.getApplicantCaseName();
+        this.applicantName = getApplicantNames(caseData);
+    }
+
+    private static String getApplicantNames(CaseData caseData) {
+        List<PartyDetails> applicants = ElementUtils.unwrapElements(caseData.getApplicants());
+        String applicantNames = "";
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/models/dto/notify/CaseInviteEmail.java:36]: Can we move this to stringBuffer please?

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java (hunk 1 above)
Comment (pete-lowe) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java:88]: You're correct, moved the controllers to a separate class - I have removed this.

File: src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/serviceofapplication/OrdersToServeSA.java (hunk 3 above)
Comment (pete-lowe) [reply src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/serviceofapplication/OrdersToServeSA.java:51]: reformatted.

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/notify/CaseInviteEmail.java (hunk 4 above)
Comment (pete-lowe) [reply src/main/java/uk/gov/hmcts/reform/prl/models/dto/notify/CaseInviteEmail.java:36]: Updated to use StringBuilder - hope thats ok?

File: src/main/java/uk/gov/hmcts/reform/prl/enums/serviceofapplication/TransferOfCaseToAnotherCourtEnum.java (hunk 2 above)
Comment (pete-lowe) [reply src/main/java/uk/gov/hmcts/reform/prl/enums/serviceofapplication/TransferOfCaseToAnotherCourtEnum.java:25]: @BharadwajSairam - can you take a look at this and remove from your service of application implementation?

**Pull request merged**